patches/node/build_remove_explicit_linker_call_to_libm_on_macos.patch

```diff
@@ -0,0 +1,58 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: deepak1556 <hop2deep@gmail.com>
+Date: Mon, 3 Feb 2025 21:44:36 +0900
+Subject: build: remove explicit linker call to libm on macOS
+
+/usr/lib/libm.tbd is available via libSystem.*.dylib and
+reexports sanitizer symbols. When building for asan
+this becomes an issue as the linker will resolve the symbols
+from the system library rather from libclang_rt.*
+
+For V8 that rely on specific version of these symbols
+that get bundled as part of clang, for ex:
+https://source.chromium.org/chromium/chromium/src/+/main:v8/src/heap/cppgc/platform.cc;l=93-97
+accepting nullptr for shadow_offset in `asan_get_shadow_mapping`,
+linking to system version that doesn't support this will lead to
+a crash.
+
+Clang driver eventually links with `-lSystem`
+https://github.com/llvm/llvm-project/blob/e82f93890daefeb38fe2a22ee3db87a89948ec57/clang/lib/Driver/ToolChains/Darwin.cpp#L1628-L1631,
+this is done after linking the sanitizer libraries which
+ensures right order of resolution for the symbols.
+
+PR-URL: https://github.com/nodejs/node/pull/56901
+Reviewed-By: Joyee Cheung <joyeec9h3@gmail.com>
+Reviewed-By: Chengzhong Wu <legendecas@gmail.com>
+Reviewed-By: Luigi Pinca <luigipinca@gmail.com>
+Reviewed-By: Shelley Vohr <shelley.vohr@gmail.com>
+
+diff --git a/deps/brotli/unofficial.gni b/deps/brotli/unofficial.gni
+index 5e07e106672a04508a77584c109c97a67926c858..91001fa43ea4807d061f296eaeccb7512e34863e 100644
+--- a/deps/brotli/unofficial.gni
++++ b/deps/brotli/unofficial.gni
+@@ -25,7 +25,7 @@ template("brotli_gn_build") {
+     } else if (target_os == "freebsd") {
+       defines = [ "OS_FREEBSD" ]
+     }
+-    if (!is_win) {
++    if (is_linux) {
+       libs = [ "m" ]
+     }
+     if (is_clang || !is_win) {
+diff --git a/deps/uv/unofficial.gni b/deps/uv/unofficial.gni
+index 7a73f891e3fc3261b77af97af63fca2eade49849..6edc420993fba6a5a5e20a25822d2dc459a82247 100644
+--- a/deps/uv/unofficial.gni
++++ b/deps/uv/unofficial.gni
+@@ -82,11 +82,11 @@ template("uv_gn_build") {
+       ]
+     }
+     if (is_posix) {
+-      libs = [ "m" ]
```

Comment:
Looks like GN is unhappy with this change because the other usage of `libs` is `+=` but now it's never been defined with an initial value.